tests/integration/test_mdg_generation.py

```diff
@@ -0,0 +1,464 @@
+"""
+
+Collection of tests for validating the mixed-dimensional grid generation
+
+Functionalities being tested:
+* Generation with/without fractures
+* Generation of meshes with type {"simplex", "cartesian", "tensor_grid"}
+* Generation of meshes with dimension {2,3}
+
+"""
+from typing import List
+
+import numpy as np
+import pytest
+
+import porepy as pp
+import porepy.grids.standard_grids.utils as utils
+
+
+class TestMDGridGeneration:
+    """Test suit for verifying the md-grid generation of pp.create_mdg"""
+
+    def cell_size(self) -> float:
+        return 0.5
+
+    def fracture_2d_data(self) -> List[np.array]:
```

Comment:
Please fix typing throughout this file.
```suggestion
    def fracture_2d_data(self) -> list[np.ndarray]:
```